src/core/library.c

```diff
@@ -1296,32 +1296,38 @@ QuicLibraryGetParam(
     return Status;
 }
 
-//
-// N.B Maintained and exported for backwards compatiblity with old V1 clients.
-//
 _IRQL_requires_max_(PASSIVE_LEVEL)
+_Check_return_
 QUIC_STATUS
 QUIC_API
-MsQuicOpen(
-    _Out_ _Pre_defensive_ const QUIC_API_TABLE** QuicApi
+MsQuicOpenVersion(
+    _In_ uint32_t Version,
+    _Out_ _Pre_defensive_ const void** QuicApi
     )
 {
     QUIC_STATUS Status;
     BOOLEAN ReleaseRefOnFailure = FALSE;
 
+    if (Version != 2) {
+        QuicTraceLogVerbose(
+            LibraryMsQuicOpenVersionUnsupported,
+            "[ api] MsQuicOpenVersion, Only version 2 supported");
```

Comment:
I think we should promote this to `QuicTraceEvent` with `LibraryError`. It's important.

Reply:
Agreed. Done